docfx_yaml/extension.py

```diff
@@ -134,6 +134,21 @@ class Bcolors:
     DEPRECATED: 'deprecated',
 }
 
+_SUMMARY_TYPE_BY_ITEM_TYPE = {
+    # Modules and Classes are similar types.
+    MODULE: CLASS,
+    CLASS: CLASS,
+}
+# Construct a mapping of name and content for each unique summary type entry.
+_ENTRY_NAME_AND_ENTRY_CONTENT_BY_SUMMARY_TYPE = {
```

Comment:
Can this be converted to a `dataclass`? Something like:
```
class SummaryEntry:
  name: str
  summary: str
```

For now it seems like there's just one summary type `CLASS`, so I think this information can be captured in just one data class. This would allow us to not use i-indexing below, which I think makes this more readable.

Reply:
It's a bit tricky, I need to keep track of all the entries that's been submitted already. Either we need to keep an extra list, otherwise converting this to a dataclass we'd lose being able to have a sequence of information, instead we'll have `Sequence[SummaryEntry]` and not be able to easily go through the info :/